python-large-file-upload/notecardDataTransfer.py

```diff
@@ -96,8 +97,13 @@ def _writeAndFlushBytes(self, data: io.IOBase):
             binary_helpers.binary_store_reset(self._card)
             rsp = self._sendRequest("card.binary")
             max = rsp.get("max", 0)
-            
-            buffer = bytearray(max)
+
+            if self._binaryBuffSize is not None:
+                buffSize = self._binaryBuffSize

```

Comment:
Should be the minimum of `self._binaryBuffSize` and `max`

```
buffSize = min(self._binaryBuffSize, max)
```


If the requested buffer size from the `_binaryBuffSize` is larger than the available space returned by `max` this could cause an issue.

Reply:
Not a requirement for approval.  Just for your consideration.

This section could be rewritten as:


```
...
buffSize = rsp.get("max", 0)
if self._binaryBuffSize is not None:
    buffSize = min(self._binaryBuffSize, buffSize)
buffer = bytearray(buffSize)
...
```

I personally prefer to avoid using `else` statements to set a default value if the value is known prior to performing the `if` part of the branch.

Sometimes that doesn't make sense as perhaps you only want to retrieve the default value in the `else` condition.  But in this case, we need the default (max) value in both cases, so we must retrieve it prior to the `if` condition.